llama_stack/providers/remote/vector_io/qdrant/qdrant.py

```diff
@@ -127,21 +123,86 @@ async def delete(self):
         await self.client.delete_collection(collection_name=self.collection_name)
 
 
-class QdrantVectorIOAdapter(VectorIO, VectorDBsProtocolPrivate):
+class QdrantVectorIOAdapter(OpenAIVectorStoreMixin, VectorIO, VectorDBsProtocolPrivate):
     def __init__(
         self, config: RemoteQdrantVectorIOConfig | InlineQdrantVectorIOConfig, inference_api: Api.inference
     ) -> None:
         self.config = config
         self.client: AsyncQdrantClient = None
         self.cache = {}
         self.inference_api = inference_api
+        self.vector_db_store = None
+        self.openai_vector_stores: dict[str, dict[str, Any]] = {}
+        self.files_api = None  # Files API is not available for Qdrant
 
     async def initialize(self) -> None:
         self.client = AsyncQdrantClient(**self.config.model_dump(exclude_none=True))
+        # Load existing OpenAI vector stores using the mixin method
+        self.openai_vector_stores = await self._load_openai_vector_stores()
 
     async def shutdown(self) -> None:
         await self.client.close()
 
+    # OpenAI Vector Store Mixin abstract method implementations
+    async def _save_openai_vector_store(self, store_id: str, store_info: dict[str, Any]) -> None:
+        """Save vector store metadata to Qdrant collection metadata."""
+        # Store metadata in a special collection for vector store metadata
+        metadata_collection = "openai_vector_stores_metadata"
+
+        # Create metadata collection if it doesn't exist
+        if not await self.client.collection_exists(metadata_collection):
+            await self.client.create_collection(
+                collection_name=metadata_collection,
+                vectors_config=models.VectorParams(size=1, distance=models.Distance.COSINE),
```

Comment:
I don't love having the distance metric hard coded instead of being configurable.